Neo4j.Driver/Neo4j.Driver/Internal/Connector/SocketConnection.cs

```diff
@@ -58,7 +71,11 @@ public void Init()
         {
             try
             {
-                Task.Run(() => _client.Start()).Wait();
+                var connected = Task.Run(() => _client.Start()).Wait(_connectionTimeout);
```

Comment:
Shouldn't we cancel the `_client.Start()` task if `!connected`?
Could it still remain running in the background?
I do not know for sure, reading these articles:
https://msdn.microsoft.com/en-us/library/mt674886.aspx
https://msdn.microsoft.com/en-us/library/system.threading.cancellationtokensource(v=vs.110).aspx
https://msdn.microsoft.com/en-us/library/dd235606(v=vs.110).aspx

Would be nice to have this tested as well.